### PR TITLE
Scrutinizer fixes

### PR DESCRIPTION
**Pull request by bizley**
Fixing some stuff that Scrutinizer shows. Many things are false negatives anyway.

### CODE REVIEW

File: framework/console/controllers/CacheController.php [hunk 1]
```diff
@@ -84,11 +84,11 @@ public function actionFlush()
         $foundCaches = array_keys($caches);
         $notFoundCaches = array_diff($cachesInput, array_keys($caches));
 
-        if ($notFoundCaches) {
+        if (!empty($notFoundCaches)) {
             $this->notifyNotFoundCaches($notFoundCaches);
         }
 
-        if (!$foundCaches) {
+        if (empty($foundCaches)) {
```
Comment (samdark) [inline framework/console/controllers/CacheController.php:91]:
```suggestion
        if ($foundCaches === []) {
```

File: framework/console/controllers/FixtureController.php [hunk 2]
```diff
@@ -122,7 +122,7 @@ public function actionLoad(array $fixturesInput = [])
             $foundFixtures = $this->findFixtures($fixtures);
             $notFoundFixtures = array_diff($fixtures, $foundFixtures);
 
-            if ($notFoundFixtures) {
+            if (!empty($notFoundFixtures)) {
```
Comment (samdark) [inline framework/console/controllers/FixtureController.php:125]:
```suggestion
            if ($notFoundFixtures !== []) {
```

File: framework/console/controllers/FixtureController.php [hunk 3]
```diff
@@ -198,23 +198,23 @@ public function actionUnload(array $fixturesInput = [])
...
```
Comment (samdark) [inline framework/console/controllers/FixtureController.php:201]:
```suggestion
            if ($notFoundFixtures !== []) {
```

File: framework/console/controllers/FixtureController.php [hunk 4]
```diff
@@ -198,23 +198,23 @@ public function actionUnload(array $fixturesInput = [])
...
                 $this->notifyNotFound($notFoundFixtures);
             }
         } else {
             $foundFixtures = $this->findFixtures();
         }
 
-        $fixturesToUnload = array_diff($foundFixtures, $except);
-
-        if (!$foundFixtures) {
+        if (empty($foundFixtures)) {
```
Comment (samdark) [inline framework/console/controllers/FixtureController.php:208]:
```suggestion
        if ($foundFixtures === []) {
```

File: framework/console/controllers/FixtureController.php [hunk 5]
```diff
@@ -198,23 +198,23 @@ public function actionUnload(array $fixturesInput = [])
...
             throw new Exception(
                 'No files were found for: "' . implode(', ', $fixturesInput) . "\".\n" .
                 "Check that files exist under fixtures path: \n\"" . $this->getFixturePath() . '".'
             );
         }
 
-        if (!$fixturesToUnload) {
+        $fixturesToUnload = array_diff($foundFixtures, $except);
+
+        if (empty($fixturesToUnload)) {
```
Comment (samdark) [inline framework/console/controllers/FixtureController.php:217]:
```suggestion
        if ($fixturesToUnload === []) {
```

File: framework/console/controllers/FixtureController.php [hunk 6]
```diff
@@ -225,7 +225,7 @@ public function actionUnload(array $fixturesInput = [])
 
         $fixtures = $this->getFixturesConfig(array_merge($this->globalFixtures, $fixturesToUnload));
 
-        if (!$fixtures) {
+        if (empty($fixtures)) {
```
Comment (samdark) [inline framework/console/controllers/FixtureController.php:228]:
```suggestion
        if ($fixtures === []) {
```

File: framework/console/controllers/MigrateController.php [hunk 7]
```diff
@@ -301,11 +301,9 @@ protected function truncateDatabase()
 
         // First drop all foreign keys,
         foreach ($schemas as $schema) {
-            if ($schema->foreignKeys) {
```
Comment (samdark) [inline framework/console/controllers/MigrateController.php:304]: Are you sure foreignKeys is always iterable?

Comment (bizley) [reply framework/console/controllers/MigrateController.php:304]: I've based this on docs but you are right, with Yii you never know :P Anyway, I've just run search and look at all the mentions for that and I'm 99% sure.

**Comment by samdark** [comment]
👍

**Pull request merged**